ext-dynamic-commitments.md

```diff
@@ -0,0 +1,1087 @@
+the dynamic commitment negotiation, then once both channel parties have
+irrevocably committed to a state with no HTLC outputs, new commitment signatures
+MUST be exchanged. This requires both parties to send `commitment_signed`
+messages that adhere to the new channel parameters. Once a node has received
+a `commitment_signed` in accordance with the new channel parameters, it MUST
+issue a `revoke_and_ack` as it normally would. Once both nodes have done this,
+normal channel operation is resumed.
+
+### Funding Output Change: General Protocol
+
+If a Funding Output Change is required, then new commitment signatures AND
+kickoff signatures MUST be exchanged. To accomplish this, the following steps
+are taken:
+
+1. Build kickoff transaction
+1. Build commitment transaction that spends kickoff output
+1. Issue a `commitment_signed` message _according to new channel parameters_
+1. Upon receipt of the remote party's `commitment_signed` message, issue a
+`kickoff_sig` message.
+1. Upon receipt of the remote party's `kickoff_sig` message, issue a
+`revoke_and_ack` for the _final commitment_ built off of the _original funding
+output_.
+
+#### Message flow to upgrade a channel to simple-taproot:
+
+        +-------+                               +-------+
+        |       |--(1)---- commit_signed------->|       |
+        |       |                               |       |
+        |       |<-(2)---- commit_signed -------|       |
+        |       |                               |       |
+        |       |                               |       |
+        |       |<-(3)----- kickoff_sig --------|       |
+        |   A   |                               |   B   |
+        |       |--(4)----- kickoff_sig ------->|       |
+        |       |                               |       |
+        |       |                               |       |
+        |       |--(5)---- revoke_and_ack ----->|       |
+        |       |                               |       |
+        |       |<-(6)---- revoke_and_ack ------|       |
+        +-------+                               +-------+
+
+##### Rationale
+
+The commitment signed message has to be issued first to ensure that the money
+locked to the new funding output (created by the kickoff transaction) can be
+unilaterally recovered. If the `kickoff_sig` were sent first, the receiver could
+stop responding and broadcast the kickoff transaction, burning the funds for
+both parties. If the channel balance is overwhelmingly imbalanced towards the
+side issuing the `kickoff_sig`, this could be costly to the victim while being
+comparatively cheap for the attacker.
+
+Similarly, if we `revoke_and_ack` prior to receiving a `kickoff_sig` then we
+may have a situation where we remove our ability to broadcast the old commitment
+transaction before the path to the new commitment transaction has been fully
+signed.
+
+#### Building the Kickoff Transaction
+
+##### Kickoff Transaction Structure
+
+* version: 2
```

Comment:
Depending on deployment timeline, it _could_ make sense to make this a V3 transaction with a single shared key anchor _if_ we forsee the kickoff being spent with its anchor _before_ the commitment is broadcast. 

This would help with pinning because sibling eviction allows us to have as many spendable outputs as we like (we don't run into "just one more" issues) , but has the (very) big downside that the commitment tx can't be broadcast with it because unconfirmed V3 parents required an unconfirmed V3 child (which the commitment isn't, but the sweep could be).

Reply:
> Depending on deployment timeline, it could make sense to make this a V3 transaction with a single shared key anchor if we forsee the kickoff being spent with its anchor before the commitment is broadcast.

Shared key anchor seems like an interesting idea. I need to get smart on the details for V3 txs before I can give you a principled stance.

> ...but has the (very) big downside that the commitment tx can't be broadcast with it because unconfirmed V3 parents required an unconfirmed V3 child (which the commitment isn't, but the sweep could be).

fwiw, there's nothing that *fundamentally* prevents a commitment tx under this scheme from being V3, but it admittedly probably makes implementation complexity very unattractive. It would require conflating "rendering" with "funding output conversion" and I *really like* that I can think about them independently right now.